scripts/dns-any-query.nse

```diff
@@ -0,0 +1,159 @@
+local stdnse = require "stdnse"
+local dns = require "dns"
+local shortport = require "shortport"
+local target = require "target"
+
+description = [[
+Checks if ANY dns type is allowed.
+
+Usually if server allows ANY queries, it will send more than one answers.
+During prerule phase, if <code>newtargets</code> script argument is passed
+it will find new targets to nmap scanning queue
+]]
+
+---
+-- @usage
+-- nmap --script=dns-any-query -p53 <host>
+--
+-- Script uses dns library to connect to dns server
+--
+-- @output
+-- PORT   STATE SERVICE REASON
+-- 53/udp open  domain  udp-response ttl 55
+-- | dns-any-query:
+-- |   Result: Server is likely responding to ANY query
+-- |   Authoritative: TRUE
+-- |   Found:
+-- |     		MX
+-- |     		SOA
+-- |     		TXT
+-- |     		TXT
+-- |_    		TXT
+
+author = "Erhad Husovic"
+license = "Same as Nmap--See https://nmap.org/book/man-legal.html"
+categories = {"default","safe"}
+
+local options = {}
+
+-- During prerule phase, we except two variables: dns-any-query.server to query against and
+-- dns-any-query.query-domain domain to query
+-- Also, passing additional script arg <code>newtargets</code> will add domain name and server
+-- to new targets
+prerule = function()
+  options.domain_name, options.server, options.full_output,options.newtargets = stdnse.get_script_args(
+	"dns-any-query.query-domain","dns-any-query.server","dns-any-query.full-output","newtargets"
+  )
+
+  if not options.domain_name then
+    return false
+  end
+
+  if not options.server then
+    return false
+  end
+
+  options.port = 53
+  if options.newtargets then
+    local status, err = target.add(options.domain_name,options.server)
```

Comment:
this is not the behavior i wold expect with newtargets.  I would expect targets discovered in the ANY answer to be added e.g.for a scan with domain_name=johnbond.org, i would expect the following newtargets
 * ns.johnbond.org. (from Ns and MX)
 * ns1.he.net. (from NS)
 * 199.38.182.204 (from A)
 * 2607:f740:e::f2d (from AAAA)

Reply:
you are right, it makes more sense